subprojects/core/src/integTest/groovy/org/gradle/api/internal/buildevents/AbstractBuildResultLoggerFunctionalTest.groovy

```diff
@@ -41,7 +42,7 @@ abstract class AbstractBuildResultLoggerFunctionalTest extends AbstractConsoleGr
         fails('fail')
 
         then:
-        failure.assertHasErrorOutput(failureMessage)
+        failure.hasErrorOutput(failureMessage)
```

Comment:
We shouldn't be using the `has` methods anywhere because the `has` methods have no diagnostics when the check fails. We should use the `assert` methods instead.

Reply:
Yeah, the issue here is that I want to check ansi output as well, so I need the raw error output while the existing "assert" methods check against the output with ansi chars removed.  I went ahead and created a few more "assert" methods on the fixtures so that you can do similar checks on the raw output and also get the diagnostics when they fail.  I replaced several of the usages of "result.output" and "result.error" with the new "assertRaw" methods, but I'm sure there are other places where they could be used, too.